incubator/mysqlha/templates/svc.yaml

```diff
@@ -1,44 +1,41 @@
-# Headless service for stable DNS entries of StatefulSet members.
-apiVersion: v1
-kind: Service
-metadata:
-  name: {{ template "fullname" . }}
-  labels:
-    app: {{ template "fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
-spec:
-  ports:
-  - name: {{ template "fullname" . }}
-    port: 3306
-  clusterIP: None
-  selector:
-    app: {{ template "fullname" . }}
----
 # Client service for connecting to any MySQL instance for reads.
-# For writes, you must instead connect to the master: mysql-0.mysql.
+{{- if .Values.service.type }}
+{{- $fullname := include "fullname" . }}
+{{- $rc_number := int .Values.mysqlha.replicaCount }}
+{{- $root := . }}
+{{ range $i := until $rc_number }}
+---
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "fullname" . }}-readonly
+  name: {{ template "fullname" $root }}-svc-{{ $i }}
```

Comment:
The intent was to connect to this service only for reads as writes always need to go through the primary. As such, I think we should keep `readonly` in the name as well as the above comment explaining this.

Reply:
Your mean like below? 

```name: {{ template "fullname" $root }}-readonly-{{ $i }}```